File: LeetCode/Easy/0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        # result = []
        # result.append([1])
        # for i in range(numRows - 1):
        #     newRow = [1]
        #     for j in range(i):
        #         newRow.append(result[i][j]+result[i][j+1])
        #     newRow.append(1)
        #     result.append(newRow)
        # return result
        result = []
        for i in range(numRows):
            newRow = [1] * (i+1)
            for j in range(1, i):
                newRow[j] = result[i-1][j-1] + result[i-1][j]
            result.append(newRow)
        return result

File: LeetCode/Easy/0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> result;
        result.push_back({1});
        for(int i=0; i<numRows-1;i++){
            vector<int> newRow;
            newRow.push_back(1);
            for(int j=0; j<i; j++){
                int elem = result[i][j] + result[i][j+1];
                newRow.push_back(elem);
            }
            newRow.push_back(1);
            result.push_back(newRow);
        }
        return result;
    }
};